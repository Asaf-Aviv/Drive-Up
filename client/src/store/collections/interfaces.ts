import { MovieShowcase } from '../movies/interfaces';

export interface Collection {
  id: number;
  name: string;
  overview: string;
  poster_path: string | null;
  backdrop_path: string | null;
  parts: MovieShowcase[];
}
