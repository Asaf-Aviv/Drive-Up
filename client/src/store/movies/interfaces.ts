export interface Results<T> {
  results: T;
  page: number;
  total_pages: number;
  total_results: number;
}

export interface Genre {
  id: number;
  name: string;
}

export interface Series {
  id: number;
  name: string;
  poster_path: string;
  backdrop_path: string;
}

export interface Company {
  name: string;
  id: number;
  logo_path: string | null;
  origin_country: string;
}

export interface Image {
  file_path: string;
  width: number;
  height: number;
}

export interface Video {
  id: string;
  key: string;
  name: string;
  site: string;
  type: string;
}

export interface MovieShowcase {
  poster_path: string | null;
  adult: boolean;
  overview: string;
  release_date: string;
  genre_ids: number[];
  id: number;
  original_title: string;
  original_language: string;
  title: string;
  backdrop_path: string | null;
  popularity: number;
  vote_count: number;
  video: boolean;
  vote_average: number;
}

export interface Movie {
  adult: boolean;
  backdrop_path: string | null;
  belongs_to_collection: Series | null;
  budget: number;
  genres: Genre[];
  homepage: string | null;
  id: number;
  images: {
    posters: Image[];
    backdrops: Image[];
  };
  videos: {
    results: Video[];
  };
  similar: Results<MovieShowcase[]>;
  recommendations: Results<MovieShowcase[]>;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string | null;
  production_companies: Company[];
  release_date: string;
  revenue: number;
  runtime: number | null;
  tagline: string | null;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}
