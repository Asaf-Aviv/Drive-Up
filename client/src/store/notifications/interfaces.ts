type NotificationVariant = 'error' | 'success' | 'info' | 'warning';

export interface Notification {
  id: string;
  variant: NotificationVariant;
  message: string;
}
