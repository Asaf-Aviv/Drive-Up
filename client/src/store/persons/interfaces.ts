export interface PersonSummary {
  profile_path: string;
  adult: boolean;
  id: number;
  name: string;
  popularity: number;
  known_for_department: string;
  gender: 0 | 1 | 2;
}
