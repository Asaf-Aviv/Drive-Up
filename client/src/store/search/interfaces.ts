import { MovieShowcase } from '../movies/interfaces';
import { PersonSummary } from '../persons/interfaces';
import { ShowShowcase } from '../shows/interfaces';

interface MediaType<T = 'movie' | 'tv' | 'person'> {
  media_type: T;
}

export type SearchResults =
  | MovieShowcase & MediaType<'movie'>
  | ShowShowcase & MediaType<'tv'>
  | PersonSummary & MediaType<'person'>;
