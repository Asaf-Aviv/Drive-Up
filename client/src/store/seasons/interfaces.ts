interface Episode {
  air_date: string;
  episode_number: number;
  guest_stars: {
    character: string;
    credit_id: string;
    id: number;
    name: string;
    profile_path: string | null;
  }[];
  name: string;
  overview: string;
  id: number;
  season_number: number;
  still_path: string | null;
  vote_average: number;
  vote_count: number;
  show_id: number;
  crew: {
    id: number;
    credit_id: string;
    name: string;
    department: string;
    job: string;
    profile_path: string | null;
  }[];
}

export interface Season {
  _id: string;
  id: string;
  air_date: string;
  name: string;
  overview: string;
  poster_path: string | null;
  season_number: number;
  episodes: Episode[];
}
