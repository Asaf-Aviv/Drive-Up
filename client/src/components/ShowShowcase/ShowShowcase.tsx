import React from 'react';
import { Link } from 'react-router-dom';
import { Box, Paper, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import { ShowShowcase as IShowShowcase } from '../../store/shows/interfaces';

const useStyles = makeStyles({
  paper: {
    marginBottom: 24,
  },
  link: {
    textDecoration: 'none',
    color: 'inherit',
  },
  overview: {
    maxWidth: 650,
    textAlign: 'center',
    margin: '0 auto',
    fontSize: '0.9em',
  },
});

const ShowShowcase: React.FC<IShowShowcase> = ({
  id,
  name,
  original_language,
  overview,
  poster_path,
  first_air_date,
}) => {
  const classes = useStyles();

  return (
    <Paper className={classes.paper} elevation={5}>
      <Box display="flex">
        <Box display="inline-flex" flexBasis={185}>
          <img
            style={{ alignSelf: 'flex-start' }}
            src={`http://image.tmdb.org/t/p/w185/${poster_path}`}
            alt={name}
          />
        </Box>
        <Box padding={3} width="100%">
          <Box display="flex" alignItems="center">
            <Box flex={1} component="span" paddingRight={3} fontWeight={600}>
              {`First Air Date: ${new Date(first_air_date).toLocaleDateString()}`}
            </Box>
            <Link className={classes.link} to={`/show/${id}`}>
              <Typography
                variant="h5"
                component="h2"
                align="center"
                gutterBottom
              >
                {name}
              </Typography>
            </Link>
            <Box
              flex={1}
              textAlign="right"
              component="span"
            >
              {`Language ${original_language.toUpperCase()}`}
            </Box>
          </Box>
          <Box display="flex" alignItems="center">
            <Typography className={classes.overview} variant="subtitle1">{overview}</Typography>
          </Box>
        </Box>
      </Box>
    </Paper>
  );
};

export default ShowShowcase;
