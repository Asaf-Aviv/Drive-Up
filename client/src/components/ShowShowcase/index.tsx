import ShowShowcase from './ShowShowcase';

export default ShowShowcase;
