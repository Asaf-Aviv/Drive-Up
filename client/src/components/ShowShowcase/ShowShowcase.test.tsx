import React from 'react';
import { mount, shallow } from 'enzyme';
import ShowShowcase from './ShowShowcase';
import { ShowShowcase as IShowShowcase } from '../../store/shows/interfaces';
import { renderWithProviders } from '../../test-utils';

const show: IShowShowcase = {
  backdrop_path: '/1pP0gg0iNGX06wSs19EQOvzfZIF.jpg',
  first_air_date: '1989-12-17',
  genre_ids: [16, 35],
  id: 456,
  name: 'The Simpsons',
  origin_country: ['US'],
  original_language: 'en',
  original_name: 'The Simpsons',
  overview: 'Set in Springfield',
  popularity: 160.624,
  poster_path: '/yTZQkSsxUFJZJe67IenRM0AEklc.jpg',
  vote_average: 7.1,
  vote_count: 2123,
};

describe('ShowShowcase component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ShowShowcase {...show} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render the name of the show', () => {
    const wrapper = mount(renderWithProviders(ShowShowcase, show));
    const name = wrapper.find('h2').text();
    expect(name).toEqual(show.name);
  });

  it('should render the overview of the show', () => {
    const wrapper = mount(renderWithProviders(ShowShowcase, show));
    const overview = wrapper.find('h6').text();
    expect(overview).toEqual(show.overview);
  });

  it('should render a link to the show by id', () => {
    const wrapper = mount(renderWithProviders(ShowShowcase, show));
    const href = wrapper.find('a').prop('href');
    expect(href).toEqual(`/show/${show.id}`);
  });

  it('should render the show image', () => {
    const wrapper = mount(renderWithProviders(ShowShowcase, show));
    const imgSrc = wrapper.find('img').prop('src');
    const baseSrc = 'http://image.tmdb.org/t/p/w';

    expect(wrapper.exists('img')).toBeTruthy();
    expect(imgSrc).toContain(baseSrc);
    expect(imgSrc).toContain(show.poster_path);
  });
});
