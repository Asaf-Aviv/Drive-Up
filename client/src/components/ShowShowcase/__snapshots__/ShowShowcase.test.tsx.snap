// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowShowcase component renders correctly 1`] = `
"<WithStyles(ForwardRef(Paper)) className=\\"makeStyles-paper-1\\" elevation={5}>
  <Styled(MuiBox) display=\\"flex\\">
    <Styled(MuiBox) display=\\"inline-flex\\" flexBasis={154}>
      <img style={{...}} src=\\"http://image.tmdb.org/t/p/w154//yTZQkSsxUFJZJe67IenRM0AEklc.jpg\\" alt=\\"The Simpsons\\" />
    </Styled(MuiBox)>
    <Styled(MuiBox) padding={3} width=\\"100%\\">
      <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\">
        <Styled(MuiBox) flex={1} component=\\"span\\" paddingRight={3} fontWeight={600}>
          First Air Date: 12/17/1989
        </Styled(MuiBox)>
        <Link className=\\"makeStyles-link-2\\" to=\\"/show/456\\">
          <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\" align=\\"center\\" gutterBottom={true}>
            The Simpsons
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Styled(MuiBox) flex={1} textAlign=\\"right\\" component=\\"span\\">
          Language EN
        </Styled(MuiBox)>
      </Styled(MuiBox)>
      <Styled(MuiBox) display=\\"flex\\" alignItems=\\"center\\">
        <WithStyles(ForwardRef(Typography)) className=\\"makeStyles-overview-3\\" variant=\\"subtitle1\\">
          Set in Springfield
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>"
`;
